shipitscript/shipitscript/shipitapi.py

```diff
@@ -123,32 +144,83 @@ def update_status(self, name, status, rebuild_product_details=True, headers={}):
                 data='{}',
                 headers=headers,
             )
-        return res
-
-    def releases_are_disabled(self, product, channel, headers={}):
-        pass
-
-    def get_next_release_version(self, product, channel, headers={}):
-        pass
+        return resp
+
+    def get_disabled_products(self, headers={}):
+        """Method to map over the GET /disabled-products/ API in Ship-it
+
+        Returns which products are disabled for which branches
+        {
+          "devedition": [
+            "releases/mozilla-beta",
+            "projects/maple"
+          ],
+          "firefox": [
+            "projects/maple",
+            "try"
+          ]
+        }
+        """
+        resp = None
+        try:
+            resp = self._request(api_endpoint='/disabled-products', headers=headers)
 
-    def get_most_recent_shipped_revision(self, product, channel, headers={}):
-        pass
+            return resp.json()
+        except Exception:
+            log.error('Caught error while getting disabled-products', exc_info=True)
+            if resp:
+                log.error(resp.content)
+                log.error(f'Response code: {resp.status_code}')
+            raise
 
     def create_new_release(
-        self, product, repo, channel, release_name, version, revision, headers={}
+        self, product, branch, version, build_number, revision, headers={}
     ):
-        if self.releases_are_disabled(product, channel):
-            return
-        pass
-
-    def trigger_release_phase(self, product, channel, release_name, phase, headers={}):
-        """Trigger a push phase for a specific release"""
-        if self.releases_are_disabled(product, channel):
-            return
-
-        self._request(
-            api_endpoint='/releases/{}/{}'.format(release_name, phase),
-            method='PUT',
-            data=None,
-            headers=headers,
+        """Method to map over the POST /releases/ API in Ship-it"""
+        resp = None
+        data = json.dumps(
+            {
+                'product': product,
+                'branch': branch,
+                'version': version,
+                'build_number': build_number,
+                'revision': revision,
+                'partial_updates': 'auto',
+            }
         )
+        try:
+            resp = self._request(
+                api_endpoint='/releases', method='POST', data=data, headers=headers,
+            )
+            return resp.json()
+        except Exception:
+            log.error('Caught error while creating the release', exc_info=True)
+            if resp:
+                log.error(resp.content)
+                log.error(f'Response code: {resp.status_code}')
+            raise
+
+    def trigger_release_phase(self, release_name, phase, headers={}):
```

Comment:
Aren't `headers` mandatory? Maybe don't use the default `{}` here?

Reply:
I believe they are mandatory everywhere because of https://github.com/mozilla-releng/scriptworker-scripts/blob/master/shipitscript/shipitscript/utils.py#L32. If so, I believe I should address this param to be mandatory for all of the functions, not just `trigger_release_phase`, wouldn't I?